### PR TITLE
Collection tests

### PR DESCRIPTION
**Pull request by mstokercricut**
* Tests similar to Primitives, but for Collections.
* Could use more tests related to mutability for collections...
* Also fixes a bug related to structs exported by reference that are marked Equatable, including tests

### CODE REVIEW

File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package com.cricut.testapi
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+
+internal class CollectionTests {
+    @Test
+    fun testCollectionValues() {
+        assertEquals(Collections.arrayOfInt, arrayListOf(2,7,3,5,8).map { it.toLong() })
```
Comment (cobbal) [inline integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt]: use `1L` for long literals.

File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            3.toLong() to 30.toLong(),
+            4.toLong() to 40.toLong(),
+            5.toLong() to 50.toLong()
+        ))
+    }
+
+    @Test
+    fun testOptionalCollectionValues() {
+        assertEquals(Collections.maybeArrayOfInt, arrayListOf(2,7,3,5,8).map { it.toLong() })
+        assert(Collections.maybeSetOfInt?.containsAll(setOf(5,9,2,4,3).map { it.toLong() }) ?: false)
```
Comment (cobbal) [inline integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt]: `assertEquals(thing, true)` is the preferred way to deal with `Bool?`

File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            5.toLong() to 50.toLong()
+        ))
+    }
+
+    @Test
+    fun testOptionalCollectionOfOptionalValues() {
+        assertEquals(Collections.maybeArrayOfMaybeInt, arrayListOf(null, 2,7,3,5,8).map { it?.toLong() })
+        assert(Collections.maybeSetOfMaybeInt?.containsAll(setOf(null, 5,9,2,4,3).map { it?.toLong() }) ?: false)
+        assertEquals(Collections.maybeDictionaryOfMaybeIntToMaybeInt, mapOf(
+            null to 100,
```
Comment (cobbal) [inline integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt]: Huh... didn't even know that was a thing in java.

File: integration-tests/TestAPI/Sources/TestAPI/Collections.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        public var boolSet: Set<Bool> = [false,true]
+        public var boolDictionary: [Bool:Bool] = [true:false, false:true]
+        public var integerArray: [Int] = [2,7,3,5,8]
+        public var integerSet: Set<Int> = [5,9,2,4,3]
+        public var integerDictionary: [Int:Int] = [1:10, 2:20, 3:30, 4:40, 5:50]
+        public var stringArray: [String] = ["apple","pear","strawberry"]
+        public var stringSet: Set<String> = ["apple","pear","strawberry"]
+        public var stringDictionary: [String:String] = ["fruitOne":"apple", "fruitTwo":"pear", "fruitThree":"strawberry"]
+        
+        public init(boolArray: [Bool] = [false,true], boolSet: Set<Bool> = [false,true], boolDictionary: [Bool : Bool] = [true:false, false:true], integerArray: [Int] = [2,7,3,5,8], integerSet: Set<Int> = [5,9,2,4,3], integerDictionary: [Int : Int] = [1:10, 2:20, 3:30, 4:40, 5:50], stringArray: [String] = ["apple","pear","strawberry"], stringSet: Set<String> = ["apple","pear","strawberry"], stringDictionary: [String : String] = ["fruitOne":"apple", "fruitTwo":"pear", "fruitThree":"strawberry"]) {
```
Comment (cobbal) [inline integration-tests/TestAPI/Sources/TestAPI/Collections.swift]: Clearly I need to set up swiftlint for this project :D

**Review by cobbal — CHANGES_REQUESTED** [review]
Good to see the TODOs. I'll take a look at them later.

File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt (hunk 1 above)
Comment (mstokercricut) [reply integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt]: Ah. Much better. I knew it couldn't be that hard.

File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt (hunk 3 above)
Comment (mstokercricut) [reply integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt]: I was surprised too. I wrote the test, and it passed, so... I guess it is.

File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt (hunk 2 above)
Comment (mstokercricut) [reply integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/CollectionTests.kt]: Much nicer.

File: integration-tests/TestAPI/Sources/TestAPI/Collections.swift (hunk 4 above)
Comment (mstokercricut) [reply integration-tests/TestAPI/Sources/TestAPI/Collections.swift]: As long as all the generated code passes too... 😎

**Review by cobbal — APPROVED** [review]

**Pull request merged**